src/main/java/org/jitsi/videobridge/AbstractEndpointMessageTransport.java

```diff
@@ -46,8 +46,20 @@
      * information.
      */
     protected final @NotNull Logger logger;
-    private Map<String, VideoConstraints> pinnedEndpointConstraintsMap;
-    private Map<String, VideoConstraints> selectedEndpointConstraintsMap;
+
+    /**
+     * The last pinned endpoints set signaled by the client, converted into
+     * a map of endpoint id -> video constraints. We need this for backwards
+     * compatibility (see more bellow in {@link #onVideoConstraintsChangedEvent()}
+     */
+    private Map<String, VideoConstraints> pinnedVideoConstraintsMap;
```

Comment:
Do we want to provide backwards compatibility for pinned endpoints or do we decide to axe it?

The state is part of the compatibility layer that handles cases like this: a pinned endpoint comes on-stage and then off-stage again. It's still a pinned endpoint.

With the new video constraints this would client side work: The client would send 180p video constraint to "pin" an endpoint A, then a 720p video constraint when endpoint A becomes selected (i.e. comes on-stage) and then a 180p video constraint again when endpoint A goes off-stage (but the user still would like to see endpoint A in the thumbnails).

We don't have these mechanics in the client right now (nor people who rely on selected and pinned messages) but we can chose to not provide backwards compatibility for pinned endpoints (as we've dropped them a while ago).

Reply:
Thanks for explaining the scenario.  It still seems like the message transport is a weird place to keep this state, could we have this compatibility logic in Endpoint?

Also, do we have a plan for how we'd implement this behavior in the new scheme?

Reply:
> We don't have these mechanics in the client right now (nor people who rely on selected and pinned messages) but we can chose to not provide backwards compatibility for pinned endpoints (as we've dropped them from Jitsi Meet a while ago, not sure when).

Oh, really?  If so then I think we can avoid the backwards compatibility in the bridge for pinned and we'll just do some new scheme when we bring it back.

Reply:
We've removed the complexity from the transport and encapsulated all the backwards compatibility logic in a separate component, hopefully making things a bit clearer. I'll mark this conversation as resolved now.